### DISCUSSION TITLE
Add attribute/class

### DISCUSSION THREAD

**Discussion by mustafa0x** [remarkjs/remark#829] [Q&A] [answered]
Is micromark/remark able to do the following: `[hi]{.hello}` ➔ `<span class=hello>hi</span>`? Similar to pandoc and markdown-it-attrs.

I saw micromark-extension-directive, but that's mostly useful for adding content, not for semantically marking existing content.

**Comment by ChristianMurphy** [comment, marked as answer]
> Is micromark/remark able to do the following: `[hi]{.hello}` ➔ `<span class=hello>hi</span>`? Similar to pandoc and markdown-it-attrs.

https://github.com/arobase-che/remark-attr as similar (if not the same) syntax, though it may not work with the latest version of remark https://github.com/arobase-che/remark-attr/issues/22

> I saw micromark-extension-directive, but that's mostly useful for adding content, not for semantically marking existing content.

Directives can be used to add content or to annotate existing content.
Using https://github.com/remarkjs/remark-directive along side https://github.com/syntax-tree/unist-util-find-all-before would allow for annotating the node/content before.
They'd be wrapped up in a plugin (introduced here https://unifiedjs.com/learn/guide/create-a-plugin/) and various approaches for working with/updating the content/AST are introduced at https://unifiedjs.com/learn/

**Reply by mustafa0x** [reply]
Thanks Christian! Yep, remark-attr is what I'm looking for, if only it were updated.

> Directives can be used to add content or to annotate existing content.

👍 

I guess I could use the directive syntax, even though it's a bit more verbose, but the issue is my documents also need to be pandoc-compatible, and the directive syntax isn't.

Thanks.

**Reply by wooorm** [reply]
See also: https://twitter.com/rauschma/status/1430458279096893440

**Reply by mustafa0x** [reply]
Yep, I'm in [the same boat](https://twitter.com/rauschma/status/1430553661839745032), I need PDF generation, so pandoc compat is a must. I also don't use rehype nor remark since micromark does what I want and I don't understand exactly what those two add.

**Reply by wooorm** [reply]
docraptor is also good at PDFs.

Those two add *a ton* — not required, but can help you do many awesome things for books!
